libs/sdk-common/src/input_parser.rs

```diff
@@ -227,6 +273,63 @@ pub async fn parse(input: &str) -> Result<InputType> {
     Err(anyhow!("Unrecognized input type"))
 }
 
+/// Parse input using provided external parsers.
+async fn parse_external(
+    input: &str,
+    external_input_parsers: &[ExternalInputParser],
+) -> Result<InputType> {
+    for parser in external_input_parsers {
+        // Check regex
+        let re = Regex::new(&parser.input_regex).context("Couldn't parse regex")?;
```

Comment:
NIT: perhaps also it is useful to add the provider_id to the error?

Reply:
Added in [86ee6e0](https://github.com/breez/breez-sdk-greenlight/pull/1143/commits/86ee6e0f9670116239d543b98576c493c5008870)